### DISCUSSION TITLE
HX711 + 4x50kg weight cells with over 10kg weights

### DISCUSSION THREAD

**Discussion by boozeman** [arendst/Tasmota#14010] [Ideas]
Hi,

I have HX711 + 4  pieces of 50kg load cells (max 200kg) connected with classic wheatstone bridge on ESP32. Calibration succeeded and the weight is somewhat accurate but if weight goes over 10.0kg the readings went 0.000kg. WeightMax is set to 180kg

```
sensor34
18:46:22.150 RSL: RESULT = {"Sensor34":{"WeightRef":1018,"WeightCal":10,"WeightMax":180000,"WeightItem":0.0,"WeightChange":"OFF","WeightDelta":100}}
```

Maybe something that I miss again. Seems like default weight sensor is like 10kg?

EDIT:

Something weird happen when scale reach out 10.5Kg, see console logs. See what happens WeightRaw reading. At first it grows but when reach 10.4, WeightRaw drops about 25000 and when I add weight it drops further. When remove extra weight the Weightraw come back to 830k. 

I double check the wheat stone bridge connection and it is like [this](https://www.researchgate.net/figure/A-schematic-drawn-in-Fritzing-representing-how-the-four-half-Wheatstone-bridge-load_fig3_349003072)

```
15:43:09.878 RSL: SENSOR = {"Time":"2021-12-12T15:43:09","HX711":{"Weight":7.3,"WeightRaw":834488},"ESP32":{"Temperature":31.1},"TempUnit":"C"}
15:43:27.526 RSL: SENSOR = {"Time":"2021-12-12T15:43:27","HX711":{"Weight":10.4,"WeightRaw":837644}}
15:43:29.854 RSL: STATE = {"Time":"2021-12-15:43:29.885 RSL: SENSOR = {"Time":"2021-12-12T15:43:29","HX711":{"Weight":10.4,"WeightRaw":837644},"ESP32":{"Temperature":31.1},"TempUnit":"C"}
15:43:49.842 RSL: STATE = {"Time":"2021-12-15:43:49.880 RSL: SENSOR = {"Time":"2021-12-12T15:43:49","HX711":{"Weight":10.4,"WeightRaw":837645},"ESP32":{"Temperature":31.1},"TempUnit":"C"}
15:44:01.826 RSL: SENSOR = {"Time":"2021-12-12T15:44:01","HX711":{"Weight":0.0,"WeightRaw":25017}}
15:44:07.832 RSL: SENSOR = {"Time":"2021-12-12T15:44:07","HX711":{"Weight":0.0,"WeightRaw":1708}}
15:44:09.817 RSL: STATE = {"Time":"2021-12-15:44:09.855 RSL: SENSOR = {"Time":"2021-12-12T15:44:09","HX711":{"Weight":0.0,"WeightRaw":2196},"ESP32":{"Temperature":31.1},"TempUnit":"C"}
15:44:19.959 RSL: SENSOR = {"Time":"2021-12-12T15:44:19","HX711":{"Weight":10.5,"WeightRaw":837679}}
15:44:21.959 RSL: SENSOR = {"Time":"2021-12-12T15:44:21","HX711":{"Weight":0.0,"WeightRaw":2561}}
15:44:29.856 RSL: STATE = {"Time":"2021-12-15:44:29.891 RSL: SENSOR = {"Time":"2021-12-12T15:44:29","HX711":{"Weight":10.5,"WeightRaw":837673},"ESP32":{"Temperature":31.1},"TempUnit":"C"}
15:44:31.561 RSL: SENSOR = {"Time":"2021-12-12T15:44:31","HX711":{"Weight":10.5,"WeightRaw":837673}}

```

EDIT2

Can anybody check this out too?

I was testing this with Olav Kallhovd [HX711_ADC](https://github.com/olkal/HX711_ADC) library and my Wheat Stone Bridge setup working just fine and accurately up to tested 80kg.

Flashed Tasmota32 Solo1 firmware with erasing the device. The same problem. I Can not measure weights over 10kg. 

Calibrated as instructions says:

Sensor34 1
Sensor34 3 <tested various calibration weights 50 to 2940g>
Sensor34 2 

Max weight is set 200kg with sensor34 5 200000

Scale also shows 220-240g too much. (reference weight 2940g shows as 3169g

EDIT3

I considered this a bug. 

HX711 works fine with ESP8266 compiled with `#define USE_HX711`  at user_config_override.h. But with ESP32-D0WDQ6 no matter what firmware I use (pre-compiled or my own build) the scale works at max 10.5kg.

**Comment by ascillato2** [comment]
Since you haven't had here any answer yet, have you tried in the Discord [Chat](https://discord.gg/Ks2Kzd4)?

**Reply by boozeman** [reply]
Hi,

Yeah. I try in Discord too. Seems like I am the only one to suffer this. Maybe I have to make this an issue.

![image](https://user-images.githubusercontent.com/40159019/147259009-72bcc69b-4478-4184-8ef9-a83226295f38.png)
This is what happen when I use the scale in ESP32. When weight goes up to 10.5kg (I hop to the scale) the WeightRaw drops about 1/10 of empty scale value and the value shows 0.000kg

Empty scale WeightRaw is 751999 and  example 3.775kg of junk WeightRaw is 755738, so it grows until it get that 10.5Kg 

Meanwhile the same Wheat Stone Bridge scale with the same HX711 works just fine in ESP8266. The only annoying thing about ESP8266 is that "auto tare" when reboot the device (ESP32 "remembers" the weight after reboot). I have to use WeightRaw and do some Home Assistant template magic to get readings right.

With ESP32 I was using GPIO19 as SCK and GPIO23 as DAT
With 8266 GPIO5 as SCK and GPIO4 as DAT


My goal here is to make reliable Watertank level measuring with scale.

**Comment by boozeman** [comment]
Hi,

Looks like I need to use the esp8266 circuit with the HX711. The problem is still relevant if I use ESP32. WeightRaw is over 800,000 and over 10kg weighs zero readings and WeighRaw drops somewhere around 28,000.

No changes or fixes to the driver? The 8266 also has an annoying tare reset if the 8266 is restarted.

Am I the only one with this problem?

**Reply by rj10rd** [reply]
Wich type of hx711 are you using
Green pcb or any other
Have u tried with esphome
Because i have green hx711 
But with tasmota is show only one time correct weight and with esphome it doesn't work

**Comment by boozeman** [comment]
Hi,

2 different types of green pcb and one red sparkfun one. Effect is the same.

The same 4x50kg wheat stone bridge with the same HX711 in:

ESP32 --> WeightRaw is over 800000 without load and over 10.5kg loads WeightRaw drops about 25000 and weight shows 0.000kg
ESP8266 --> All work just fine, but there is that annoying autotare if reset of power loss. (Useless when measuring constant weight eg. Water tank)

I give up with this and use MPX pressure sensor with that tank

**Comment by dingausmwald** [comment]
This is strange behaviour. I just got the same setting with esp32 to work, but i used a different wiring, according to this video https://www.youtube.com/watch?v=ofHqZJTxBnE
your raw readings seem a bit high to me. I got readings at about 4 or 5 digits. Readings like yours with 6 digits i only got with wrong/different wiring, with an additional unwanted drift over time.

Now, with the wiring from the video i got stable readings (+-500g in 12 hours). But its not working perfect after all. When calibrating with low weights, the readings are off at higher weights. When i calibrate with high weights, like a beer box (20kg), the lower weights are okay, but higher weights (like beer box + me sitting on it, 20+75kg) are not correct (60kg). Its difficult to calibrate too. The calibration is to fast. I assume that the bump from the beer box gets red so i have to place it really soft to get accurate calibration. So calibrating with even higher weights, like myself gets quite impossible since i would have to stay really still in the 3sec calibration time. There is a little chance that my used material (3D printed pods for the weightcells, https://cbrell.de/blog/waegezelle-wie-funktionieren-preiswerte-personenwaagen-zellen/) are the couse for my bad readings.

In addition, Sensor34 3 <refrence weight> has no other effect than changing the value for the reference weight. The raw and actual readings stay the same. Only Sensor34 4 <calibrated scale value> has an effect. But there seems to be more to the calibration.

You are really saying that everything works as expected with esp8266 and tasmota? Thats the last test i didn't do.
I've red this in your link: "Enable delay for writing to sck pin. This could be required for faster mcu's like the ESP32 (default: no delay)"

I'd really like to get this working with esp32 (i use a wemos d1mini esp32 clone) and the cheap green hx711. Anybody any suggestions or a working setup with esp32?

**Reply by dexterbot80** [reply]
Hello @dingausmwald
Hello @boozeman 

Did you manage to calibrate correctly in the end?
If I set the calibrated scale value at 1139 I get 0.752 and if I multiply by 100 I get exactly my weight 75.2 KG.
I think that if we could enter a value with decimals to the calibrated scale value, we could calibrate much more correctly.
in my case 11.39 would be the good value for calibrated scale value.

**Reply by boozeman** [reply]
> Hello @dingausmwald Hello @boozeman
> 
> Did you manage to calibrate correctly in the end? If I set the calibrated scale value at 1139 I get 0.752 and if I multiply by 100 I get exactly my weight 75.2 KG. I think that if we could enter a value with decimals to the calibrated scale value, we could calibrate much more correctly. in my case 11.39 would be the good value for calibrated scale value.

Yeah, it worked and now constant use since last summer. 

Check f your max value is set correct. 
sensor34 5 <max weight in grams>
If you have problems with calibration, please check that the scale is exactly in level. I was some problems with calibration and discovered that one of the four pad needs about 0.8mm shim.

I now use Sparkfun Cylinder type four wire 50kg gauges. Those cheap chinense gauges can be thrown to nearest SER-dump :)

Cheers

Timo

**Comment by dexterbot80** [comment]
Hello @boozeman 
Thank you for your fast response.
`11:14:12.176 MQT: stat/tasmota_D3BFBC/RESULT = {"Sensor34":{"Ref":2000,"Cal":1139,"Max":10000000,"Item":0.0,"Change":"OFF","Delta":0,"Tare":0,"AbsConvA":0,"AbsConvB":0}}`

Although I entered sensor34 5 100000 for 100kg, the above value appears

`11:14:53.041 MQT: tele/tasmota_D3BFBC/SENSOR = {"Time":"2024-09-03T11:14:53","HX711":{"Weight":0.020,"WeightRaw":105,"AbsRaw":119595}}`

If I automatically calibrate using my own weight, the "calibrated scale value" is automatically set to 11 and the measurements are not accurate.

I would need to set the "calibrated scale value" to 11.39 to have accurate measurements.
I don't know how to do this - I managed to enter only 1139 and then the weight obtained for 2000 grams is exactly 0.020.

I use a weight scale made in China that can weigh up to 400KG and I can say that using the original display it is very very accurate.

**Reply by boozeman** [reply]
> Hello @boozeman Thank you for your fast response. `11:14:12.176 MQT: stat/tasmota_D3BFBC/RESULT = {"Sensor34":{"Ref":2000,"Cal":1139,"Max":10000000,"Item":0.0,"Change":"OFF","Delta":0,"Tare":0,"AbsConvA":0,"AbsConvB":0}}`
> 
> Although I entered sensor34 5 100000 for 100kg, the above value appears
> 
> `11:14:53.041 MQT: tele/tasmota_D3BFBC/SENSOR = {"Time":"2024-09-03T11:14:53","HX711":{"Weight":0.020,"WeightRaw":105,"AbsRaw":119595}}`
> 
> If I automatically calibrate using my own weight, the "calibrated scale value" is automatically set to 11 and the measurements are not accurate.
> 
> I would need to set the "calibrated scale value" to 11.39 to have accurate measurements. I don't know how to do this - I managed to enter only 1139 and then the weight obtained for 2000 grams is exactly 0.020.
> 
> I use a weight scale made in China that can weigh up to 400KG and I can say that using the original display it is very very accurate.

Hi,

This is my settings for sensor34 (Made some Home Assistant scripts for setting up those with just button click)

`14:26:50.659 MQT: greenhouse/stat/Greenhouse1/RESULT = {"Sensor34":{"Ref":3059,"Cal":19,"Max":18928000,"Item":0.0,"Change":"OFF","Delta":0,"Tare":12013,"AbsConvA":0,"AbsConvB":0}}`

As you see that Max value is really weird... It is set for 150 000 (150kg)

My reference weight is old 3kg gym weight set by `sensor34 3 3059`
Calibrating with `Sensor34 2 3059`

I reset the scale with tare (150l water tank and upper plate of the scale) every time when add new patch of water 

`sensor34 10 0`
wait 5 sec
`sensor34 10 1`

WeightRes is set to 2 `WeightRes 2`  to show kilograms/litres with 2 desimal

I have never set calibrated scale value by hand with `sensor34 4 <value>` It configured automatically when calibrating.

I am using ESP32 with self-compiled firmware, version 14.2.0.3 I have used cheap no name chinese  HX711 and also Spark Fun boards

**Reply by dexterbot80** [reply]
Yes, I agree with this, but I have 400KG weight cells and when the "calibrated scale value" is automatically calibrated, it is set to the value 11. From the tests, I noticed that the optimal value for the "calibrated scale value" must be 11,39.
Even if I enter 11,39, the value 11 remains saved - probably the code here needs to be finished to be able to enter values ​​with decimals.
![image](https://github.com/user-attachments/assets/2b745d86-1535-4002-ac20-f891c303e869)
![image](https://github.com/user-attachments/assets/71542ac5-9c0b-4310-851c-7bb21505e74a)
Here i enter the 1139 for the calibrated scale value and i multiply the result by 100 in Grafana :(

I notice that your "calibrated scale value" is a fixed number "Cal":19

**Reply by dexterbot80** [reply]
Hello @boozeman 
I saw that you use a weight of 3059 grams.
Could you film the entire calibration process and the measurements you make with that weight of 3059 grams after calibration?
You can use VSDC Screen Recorder to do that
Thank you

**Reply by boozeman** [reply]
Hi, 

I will try to remember this when I fill the water tank next time.
